client/client/src/App.tsx

```diff
@@ -5,117 +5,30 @@ import './assets/pencil-solid.svg';
-    setTargetEntryForm({
-      formTitle: '',
-      formURL: '',
-      formNotes: '',
-      entryId: 0,
-    });
-    handleViewSwap();
-  }
-  function handleEntriesViewSwap() {
-    setView('entriesView');
-    setEntriesViewClass('');
-    setFormViewClass('hidden');
-    setDeleteButtonClass('invisible');
-  }
-  function handleViewSwap() {
-    if (view === 'entriesView') {
-      setView('formView');
-      setEntriesViewClass('hidden');
-      setFormViewClass('');
-      setDeleteButtonClass('invisible');
-    } else {
-      setView('entriesView');
-      setEntriesViewClass('');
-      setFormViewClass('hidden');
-      setDeleteButtonClass('invisible');
-    }
-  }
-
-  function handleEditButtonClick(event) {
-    const targetEntry = event.currentTarget.closest('li').dataset['entryId'];
-    readEntries().forEach((element) => {
-      if (Number(targetEntry) === element.entryId) {
-        handleViewSwap();
-        setDeleteButtonClass('');
-        setTitle('Edit Entry');
-        setTargetEntryForm(element);
-      }
-    });
-  }
-
-  function handleDeleteButtonClick() {
-    setModalClass('');
-  }
+  const [isEditing, setEditing] = useState<Entry | null | undefined>();
 
   return (
     <>
-      <header className="header purple-background">
-        <div className="container">
-          <div className="row">
-            <div className="column-full d-flex align-center">
-              <h1 className="white-text">Code Journal</h1>
-              <h3>
-                <a
-                  onClick={handleEntriesViewSwap}
-                  id="entriesLink"
-                  className="entries-link white-text"
-                  href="#">
-                  Entries
-                </a>
-              </h3>
-            </div>
-          </div>
-        </div>
-      </header>
-      <EntryFormView
-        targetEntry={targetEntryForm}
-        pageTitle={title}
-        title={title}
-        formTitle={targetEntryForm.formTitle}
-        url={targetEntryForm.formURL}
-        notes={targetEntryForm.formNotes}
-        formViewClass={formViewClass}
-        deleteButtonClass={deleteButtonClass}
-        viewSwapFunction={() => handleEntriesViewSwap()}
-        onDeleteButtonClick={handleDeleteButtonClick}
-      />
-      <EntriesView
-        entriesView={entriesViewClass}
-        onNewButtonClick={handleNewButtonClick}
-        onEditButtonClick={handleEditButtonClick}
-      />
-      <DeleteEntryModal
-        deleteEntryModalClass={modalClass}
-        onCancel={() => setModalClass('hidden')}
-        onConfirm={function d() {
-          removeEntry(targetEntryForm.entryId);
-          setModalClass('hidden');
-          handleViewSwap();
-        }}
-      />
+      <NavBar onEntriesButtonClick={() => setEditing(undefined)} />
+      {isEditing !== undefined ? (
+        <EntryFormView
+          targetEntry={isEditing}
+          handleViewSwap={() => {
+            setEditing(undefined);
+          }}
```

Comment:
You can also just pass `setEditing` instead as a prop instead of `handleViewSwap` and this anonymous function. 

Passing `setEditing` is more logical since we are no longer telling the app to "swap views", in an React app, we are updating the state, then React automatically handles the rendering based on the declared state. re: https://react.dev/learn/reacting-to-input-with-state#recap